extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * Script migrates the user avatars from DFS to user avatars service
+ *
+ * @see PLATFORM-1419
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of a community wiki
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class AvatarsMigratorException extends Exception {}
+
+/**
+ * Maintenance script class
+ */
+class AvatarsMigrator extends Maintenance {
+
+	public function execute() {
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		global $wgExternalSharedDB;
+		$db = $this->getDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
+
+		// process users
+		foreach ( $res as $row ) {
+			try {
+				$this->processUser( User::newFromId( $row->id ) );
+			}
+			catch ( AvatarsMigratorException $e ) {
+				\Wikia\Logger\WikiaLogger::instance()->error( __CLASS__, [
+					'exception' => $e,
+					'user_id'   => $user->getId(),
+				] );
+
+				$this->output( sprintf( "\n\t%s", $e->getMessage() ) );
+			}
+		}
+
+		$this->output( "Processing {$res->numRows()} users...\n" );
+	}
+
+	/**
+	 * Migrate the avatar of a given user
+	 *
+	 * @param User $user
+	 * @throws AvatarsMigratorException
+	 */
+	private function processUser( User $user ) {
+		$this->output( sprintf( "\n[User #%d]: ", $user->getId() ) );
+
+		$avatar = $user->getGlobalAttribute( AVATAR_USER_OPTION_NAME );
+
+		// no avatar set, skip this account
+		if ( is_null( $avatar ) ) {
+			$this->output( 'no avatar set - skipping' );
+			return;
+		}
+
+		// no need to store default avatars in user properties - remove the entry
+		else if ( self::isDefaultAvatar( $avatar ) ) {
+			$this->output( 'default avatar set - removing an attribute' );
+			$user->removeGlobalPreference( AVATAR_USER_OPTION_NAME );
```

Comment:
@jsutterfield - can I call `User::removeGlobalPreference` method to remove the global attribute?

It makes no sense to keep the default avatar in the user preferences database (142k rows currently).